contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+    }
+
+    /// @dev Sets the LicensingModule address.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param newLicensingModule The address of the LicensingModule
+    function setLicensingModule(address newLicensingModule) external onlyProtocolAdmin {
+        if (newLicensingModule == address(0)) {
+            revert Errors.LicenseRegistry__ZeroLicensingModule();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.licensingModule = ILicensingModule(newLicensingModule);
+    }
+
+    function setDefaultLicenseConfig(
+        address newLicenseTemplate,
+        uint256 newLicenseConfigId
+    ) external onlyProtocolAdmin {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.defaultLicenseTemplate = newLicenseTemplate;
+        $.defaultLicenseConfigId = newLicenseConfigId;
+    }
+
+    function registerLicenseTemplate(address licenseTemplate) external onlyProtocolAdmin {
+        if (licenseTemplate.supportsInterface(type(ILicenseTemplate).interfaceId)) {
+            revert Errors.LicenseRegistry__NotLicenseTemplate(licenseTemplate);
+        }
+        _getLicenseRegistryStorage().registeredLicenseTemplates[licenseTemplate] = true;
+        emit LicenseTemplateRegistered(licenseTemplate);
+    }
+
+    function registerRoyaltyPolicy(address royaltyPolicy) external onlyProtocolAdmin {
+        _getLicenseRegistryStorage().registeredRoyaltyPolicies[royaltyPolicy] = true;
+        emit RoyaltyPolicyRegistered(royaltyPolicy);
+    }
+
+    function registerCurrencyToken(address token) external onlyProtocolAdmin {
+        _getLicenseRegistryStorage().registeredCurrencyTokens[token] = true;
+        emit CurrencyTokenRegistered(token);
+    }
+
+    function setExpireTime(address ipId, uint256 expireTime) external onlyLicensingModule {
+        _setExpireTime(ipId, expireTime);
+    }
+
+    function setMintingLicenseSpec(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseConfigId,
+        Licensing.MintingLicenseSpec calldata mintingLicenseSpec
+    ) external onlyLicensingModule {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if (!$.registeredLicenseTemplates[licenseTemplate]) {
+            revert Errors.LicenseRegistry__UnregisteredLicenseTemplate(licenseTemplate);
+        }
+        $.mintingLicenseSpecs[_getHash(ipId, licenseTemplate, licenseConfigId)] = Licensing.MintingLicenseSpec({
+            isSet: true,
+            mintingFee: mintingLicenseSpec.mintingFee,
+            mintingFeeModule: mintingLicenseSpec.mintingFeeModule,
+            receiverCheckModule: mintingLicenseSpec.receiverCheckModule,
+            receiverCheckData: mintingLicenseSpec.receiverCheckData
+        });
+
+        emit MintingLicenseSpecSet(ipId, licenseTemplate, licenseConfigId, mintingLicenseSpec);
+    }
+
+    function setMintingLicenseSpecForAll(
```

Comment:
What is the purpose of `mintinLicenseSpecForAll`? Is this some kind of default spec?
Thanks

Reply:
Yes, it is for setting the default configuration for all licenses under the given IP.